### PR TITLE
publish: Read `keywords` metadata from embedded `Cargo.toml` file

### PR DESCRIPTION
**Pull request by Turbo87**
... instead of the metadata JSON blob

see also:

- https://github.com/rust-lang/crates.io/pull/7194
- https://github.com/rust-lang/crates.io/pull/7201
- https://github.com/rust-lang/crates.io/pull/7204
- https://github.com/rust-lang/crates.io/pull/7209

### CODE REVIEW

File: src/controllers/krate/publish.rs [hunk 1]
```diff
@@ -157,6 +157,25 @@ pub async fn publish(app: AppState, req: BytesRequest) -> AppResult<Json<GoodCra
...
+        let keywords = package
+            .keywords
+            .map(|it| it.as_local().unwrap())
+            .unwrap_or_default();
+
+        if keywords.len() > 5 {
+            return Err(cargo_err("expected at most 5 keywords per crate"));
+        }
+
+        for keyword in keywords.iter() {
```
Comment (hi-rustin) [inline src/controllers/krate/publish.rs:169]: Maybe we can extract it as a validation function. The publish function is a bit too long.

**Review by hi-rustin — APPROVED** [review]
A nit issue. Rest LGTM.

File: src/controllers/krate/publish.rs (hunk 1 above)
Comment (Turbo87) [reply src/controllers/krate/publish.rs:169]: yeah, that's the plan once I have all of the other things migrated :)

Comment (hi-rustin) [reply src/controllers/krate/publish.rs:169]: Sounds good! :shipit:

**Pull request merged**